05_iigo/index.tex

```diff
@@ -583,4 +578,254 @@ \subsubsection{Server-side}
 The monitoring cache is cleared after all roles have made a decision on whether to monitor and that decision has been carried out. If the IIGO session terminates early due to an error related to insufficient resources, the cache is not cleared. This allows wrongdoing by an IIGO role to be discovered at a later turn.
 
 \subsection{Rule Representation}
+
+\begin{quote}
+    "Learn the rules like a pro, so you can break them like an artist."
+    \linebreak
+    \noindent \emph{Pablo Picasso}
+\end{quote}
+
+Rules underpin most of the operation of the IIGO. Hence, we needed their implementation to be rather flexible in our coursework.
+Furthermore, we noticed that following rules will be difficult for agents unless we provided them with the ability to \emph{not} only check whether they are complicit with them but also with the inner workings of the rules so that agents could calculate for themselves how to become complicit if they wish so.
+Moreover, we also concluded that we should rather opt for a mathematical representation of the rules since agents would be able to much more easily inspect the relevant mathematical equation(s) more easily than a text or functional representation of the rules.
+The mathematics we decided to employ was matrices and the world of linear algebra that surrounds them.
+\linebreak
+The premise of this choice was that if a rule could be represented by matrices, then agents could look up every element of the matrix without any significant difficulty. This would enable us to effectively expose the entire operation of any rule to the agent without reducing the scope of our rules too much.
+To give an example for this matrices-based rule representation, let us consider a simple rule first. \\ ```
+\begin{rule_IIGO}
+    "If you are expected to pay $X$ amount of taxation, the amount of tax you pay must be $X$."
+\label{rule:basicTaxRule}
+\end{rule_IIGO}
+\par
+Notice that rule \ref{rule:basicTaxRule} is a rather simple one. However, it allows us to form a basis for how to represent rules as matrices.
+If our goal is simply to ensure that an agent pays the expected amount of tax, a basic mathematical operation we can perform is to subtract the actual tax paid from the expected.
+If we obtain $0$ from this subtraction, we can conclude that this agent has adhered to this rule. \\
+Formally, let $x$ and $y$ be the expected and the actual paid amount of tax, respectively. To confirm adherence to the rule, we want
+\begin{equation}
+    y - x = 0 . 
+    \label{equation:basicMaths}
+\end{equation}
+
+To turn \ref{equation:basicMaths}  into a matrix calculation, we simply have to introduce some trivial coefficients.
+\begin{equation}
+    -1*x +
+       1*y = 0
+       \label{equation:coefLinear}
+\end{equation}
+      
+Now we can write \ref{equation:coefLinear} as a matrix calculation.
```

Comment:
let's use `\eqref` for equations...
AS STATED in the guidelines :)

Reply:
Apologies